### PR TITLE
Make TPM more optional (Fixes: #2360)

### PR DESCRIPTION
**Pull request by superm1**
- Rename the `plugin_tpm` option to `tpm` and when it's disabled
remove TPM support from all plugins
- If enabled then require the TSS to be installed

Type of pull request:
- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: contrib/fwupd.spec.in [hunk 1]
```diff
@@ -424,7 +424,7 @@ mkdir -p --mode=0700 $RPM_BUILD_ROOT%{_localstatedir}/lib/fwupd/gnupg
 %{_libdir}/fwupd-plugins-3/libfu_plugin_thunderbolt.so
 %if 0%{?have_uefi}
 %{_libdir}/fwupd-plugins-3/libfu_plugin_tpm.so
-%{_libdir}/fwupd-plugins-3/libfu_plugin_tpm_eventlog.so
+%{_libdir}/fwupd-plugins-3/libfu_pluign_tpm_eventlog.so
```
Comment (hughsie) [inline contrib/fwupd.spec.in:427]: Err? Plugin?

File: meson.build [hunk 2]
```diff
@@ -226,10 +226,6 @@ libgcab = dependency('libgcab-1.0', version : '>= 1.0', fallback : ['gcab', 'gca
 gcab = find_program('gcab', required : true)
 bashcomp = dependency('bash-completion', required: false)
 python3 = find_program('python3')
-tpm2tss = dependency('tss2-esys', version : '>= 2.0', required: false)
-if tpm2tss.found()
```
Comment (hughsie) [inline meson.build:230]: Don't you still need this config.h define?

**Review by hughsie — CHANGES_REQUESTED** [review]

File: contrib/fwupd.spec.in (hunk 1 above)
Comment (superm1) [reply contrib/fwupd.spec.in:427]: Whoops :)  I'll fix it.

File: meson.build (hunk 2 above)
Comment (superm1) [reply meson.build:230]: No, take a look at rest of meson.build to see how I did it with the empty `dependency('', required:false)`

Comment (hughsie) [reply meson.build:230]: I'm not sure how that defines `ifndef HAVE_TSS2`

Comment (superm1) [reply meson.build:230]: Oh I see what you mean and my mistake.  Pushed a fix.

**Review by hughsie — APPROVED** [review]

**Pull request merged**